db/table_cache.cc

```diff
@@ -146,7 +146,8 @@ Status TableCache::GetTableReader(
             false /* force_direct_prefetch */, level, block_cache_tracer_,
             max_file_size_for_l0_meta_pin, db_session_id_,
             file_meta.fd.GetNumber(), expected_unique_id,
-            file_meta.fd.largest_seqno, file_meta.tail_size),
+            file_meta.fd.largest_seqno, file_meta.tail_size,
+            file_meta.user_defined_timestamps_persisted),
```

Comment:
There are a few more places where we use `TableReaderOptions`, for example, in ingestion and CF import. I think regarding those two in particular, neither is currently supported in conjunction with UDTS but maybe we could put TODOs there to remind us to pass the correct `user_defined_timestamps_persisted` flag once we get there.

Reply:
Thank you for this reminder! I added some TODOs for these two places. I realized that `sst_dump` also uses TableReaderOptions that needs proper support for this flag. I added the change there for now, but I think more importantly I need to make sure the tool continues to work well for this type of SST files. I created a separate task to track this.